test/e2e/testing-manifests/auth/encrypt/setup-cluster-prereqs.sh

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env bash
+
+# Copyright 2023 The Kubernetes Authors.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+set -o errexit
+set -o nounset
+set -o pipefail
+
+# build mock plugin
+buildAndPushMockPlugin() {
+    docker buildx build \
+        --no-cache \
+        --platform linux/amd64 \
+        --output=type=docker \
+        -t localhost:5000/base64:e2e \
+        -f staging/src/k8s.io/kms/internal/plugins/mock/Dockerfile staging/src/k8s.io/ \
+        --progress=plain;
+
+    docker push localhost:5000/base64:e2e
+}
+
+# create local registry
+createRegistry() {
+    running="$(docker inspect -f '{{.State.Running}}' "kind-registry" 2>/dev/null || true)"
+    if [ "${running}" != 'true' ]; then
+        echo "Creating local registry"
+        docker run \
+            -d --restart=always -p "5000:5000" --name "kind-registry" \
+            registry:2
+    else 
+        echo "Local registry is already running"
+    fi
+}
+
+# connect registry to kind network
+connectRegistry(){
+    # wait for the kind network to exist
+    for ((; ;)); do
+        if docker network ls | grep "kind"; then
+            break
+        else
+            sleep 1
+    fi
+    done
+
+    containers=$(docker network inspect "kind" -f "{{range .Containers}}{{.Name}} {{end}}")
+    needs_connect="true"
+    for c in $containers; do
+        if [ "$c" = "kind-registry" ]; then
+            needs_connect="false"
+        fi
+    done
+
+    if [ "${needs_connect}" = "true" ]; then
+        echo "connecting kind network to kind-registry"
+        docker network connect "kind" "kind-registry" || true
```

Comment:
I think we can remove this. Even if we make this pass here, the kubetest2 will ultimately fail if it can't pull the image. But I'll let @aramase  also chime in with the context when he originally wrote it.

Reply:
Talked to @aramase offline. I'll remove it.

Reply:
This can be fixed in a follow up if needed.